include/irr/builtin/shaders/loaders/gltf/vertex_impl.glsl

```diff
@@ -0,0 +1,88 @@
+#include "irr/builtin/shaders/loaders/gltf/common.glsl"
+
+#ifndef _IRR_VERT_INPUTS_DEFINED_
+#define _IRR_VERT_INPUTS_DEFINED_
+
+layout (location = _IRR_POSITION_ATTRIBUTE_ID) in vec3 vPos;
+layout(location = _IRR_NORMAL_ATTRIBUTE_ID) in vec3 vNormal;
+
+#ifndef _DISABLE_COLOR_ATTRIBUTES
+layout(location = _IRR_COLOR_ATTRIBUTE_BEGINING_ID) in vec4 vColor_0;
+#ifdef _IRR_EXTRA_COLOR_ATTRIBUTE_ID_1_ENABLE
+#define _IRR_EXTRA_COLOR_ATTRIBUTE_ID_1 _IRR_COLOR_ATTRIBUTE_BEGINING_ID + 1
+layout(location = _IRR_EXTRA_COLOR_ATTRIBUTE_ID_1) in vec4 vColor_1
+#endif // _IRR_EXTRA_COLOR_ATTRIBUTE_ID_1_ENABLE
+#ifdef _IRR_EXTRA_COLOR_ATTRIBUTE_ID_2_ENABLE
+#define _IRR_EXTRA_COLOR_ATTRIBUTE_ID_2 _IRR_COLOR_ATTRIBUTE_BEGINING_ID + 2
+layout(location = _IRR_EXTRA_COLOR_ATTRIBUTE_ID_2) in vec4 vColor_2
+#endif // _IRR_EXTRA_UV_ATTRIBUTE_ID_2_ENABLE
+#endif // _DISABLE_COLOR_ATTRIBUTES
+
+#ifndef _DISABLE_UV_ATTRIBUTES
+layout (location = _IRR_UV_ATTRIBUTE_BEGINING_ID) in vec2 vUV_0;
+#ifdef _IRR_EXTRA_UV_ATTRIBUTE_ID_1_ENABLE
+#define _IRR_EXTRA_UV_ATTRIBUTE_ID_1 _IRR_UV_ATTRIBUTE_BEGINING_ID + 1
+layout (location = _IRR_EXTRA_UV_ATTRIBUTE_ID_1) in vec2 vUV_1
+#endif // _IRR_EXTRA_UV_ATTRIBUTE_ID_1_ENABLE
+#ifdef _IRR_EXTRA_UV_ATTRIBUTE_ID_2_ENABLE
+#define _IRR_EXTRA_UV_ATTRIBUTE_ID_2 _IRR_UV_ATTRIBUTE_BEGINING_ID + 2
+layout(location = _IRR_EXTRA_UV_ATTRIBUTE_ID_2) in vec2 vUV_2
+#endif // _IRR_EXTRA_UV_ATTRIBUTE_ID_2_ENABLE
+#endif // _DISABLE_UV_ATTRIBUTES
+
+#if !defined(_DISABLE_COLOR_ATTRIBUTES) && !defined(_DISABLE_UV_ATTRIBUTES)
+#error "ERROR: UV and Color attributes must not be defined both either!"
+#endif
+
+#endif //_IRR_VERT_INPUTS_DEFINED_
+
+#ifndef _IRR_VERT_OUTPUTS_DEFINED_
+#define _IRR_VERT_OUTPUTS_DEFINED_
+
+/*
+    glTF Loader's shaders don't support multiple out attributes
+    at the moment, to change in future
+*/
+
+layout (location = 0) out vec3 LocalPos;
+layout(location = 1) out vec3 Normal;
+layout (location = 2) out vec3 ViewPos;
```

Comment:
you only ever need to output `ViewPos`, `WorldPos` or `LocalPos`... there's never a need to output 2 or more of these.